paasta_tools/deployd/watchers.py

```diff
@@ -45,6 +55,130 @@ def startup_checker(self, obj):
         self.is_ready = True
 
 
+class PublicConfigFileWatcher(PaastaWatcher):
+
+    def __init__(self, inbox_q, cluster, **kwargs):
+        super(PublicConfigFileWatcher, self).__init__(inbox_q, cluster)
+        self.wm = pyinotify.WatchManager()
+        self.wm.add_watch(PATH_TO_SYSTEM_PAASTA_CONFIG_DIR, self.mask, rec=True)
+        self.notifier = pyinotify.Notifier(watch_manager=self.wm,
+                                           default_proc_fun=PublicConfigEventHandler(filewatcher=self))
+
+    @property
+    def mask(self):
+        boring_flags = ['IN_CLOSE_NOWRITE', 'IN_OPEN', 'IN_ACCESS', 'IN_ATTRIB']
+        return reduce(lambda x, y: x | y,
+                      [v for k, v in pyinotify.EventsCodes.OP_FLAGS.items() if k not in boring_flags])
+
+    def run(self):
+        self.notifier.loop(callback=self.startup_checker)
+
+    def startup_checker(self, obj):
+        self.is_ready = True
+
+
+def get_marathon_apps():
+    marathon_config = load_marathon_config()
+    marathon_client = get_marathon_client(marathon_config.get_url(), marathon_config.get_username(),
```

Comment:
The ttl is actually also 20s https://github.com/Yelp/paasta/blob/master/paasta_tools/marathon_tools.py#L555

So unless we make it configurable it won't do much yet. I think I'm going to need to do quite a bit of work on caching soon so I think I'll leave this be for now.